### ISSUE TITLE
newFile.fd is undefined on custom upload

### ISSUE THREAD

**Issue by elennaro** [balderdashy/skipper#115]
```JavaScript
var SkipperDiskAdapter = require('skipper-disk');

...
//Inside controller upload method:
...

var receiver = SkipperDiskAdapter().receive();

req.file('file')
	.on('error', function onError(error) {
		sails.log.error("error uploading file", err);
		cb(err);
	})
	.on('finish', function onSuccess(data) {
		sails.log.info("Done uploading file", data);
		cb(null, data);
	})
	.pipe(receiver);
```

causes error:

```bash
TypeError: Cannot read property 'match' of undefined
  at Writable.onFile (...\node_modules\skipper-disk\standalone\build-disk-receiver-stream.js:91:22)
  at doWrite (_stream_writable.js:292:12)
  at writeOrBuffer (_stream_writable.js:278:5)
  at Writable.write (_stream_writable.js:207:11)
  at Upstream.ondata (_stream_readable.js:528:20)
  at emitOne (events.js:77:13)
  at Upstream.emit (events.js:169:7)
  at Upstream.Readable.read (_stream_readable.js:360:10)
  at flow (_stream_readable.js:743:26)
  at resume_ (_stream_readable.js:723:3)
  at doNTCallback2 (node.js:441:9)
  at process._tickDomainCallback (node.js:396:17)
```

**Comment by sailsbot** [comment]
Thanks for posting, @elennaro.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by Bazze** [comment]
Under `upstream.pipe` in the documentation it says:

> Also bear in mind that you must first intercept the upstream and attach an fd (file descriptor) property to each incoming file stream.

I'm currently trying to figure out how to actually do this since I can't really find any examples for it :(
